planetiler-core/src/main/java/com/onthegomap/planetiler/geo/TileExtents.java

```diff
@@ -52,9 +59,9 @@ public boolean test(TileCoord tileCoord) {
 
   /**
    * X/Y extents within a given zoom level. {@code minX} and {@code minY} are inclusive and {@code maxX} and {@code
-   * maxY} are exclusive.
+   * maxY} are exclusive. shape is an optional polygon defining a more refine shape
    */
-  public record ForZoom(int minX, int minY, int maxX, int maxY) {
+  public record ForZoom(int minX, int minY, int maxX, int maxY, MultiPolygon shape) {
```

Comment:
also let's make sure we update the tests in `TileExtentsTests` to verify the logic once it's settled

Reply:
@msbarry not sure i get the necessary test here. I added the `shape` to `ForZoom` just so that i can get it in `TiledGeometry`

Reply:
This would be after we push the test(x,y,z) logic that compares to the shape into TileExtents